lib/postzord/receiver/local_batch.rb

```diff
@@ -25,6 +25,15 @@ def receive!
     # 09/27/11 this is slow
     notify_users
 
+    if @object.respond_to?(:type) && @object.type == "Reshare"
+      root_object = Post.find_by guid: @object.root_guid
```

Comment:
It would be more helpful to say what is the alternative instead of asking me the question of why. This way we only have an extra loop to go through.